pkg/services/contexthandler/contexthandler.go

```diff
@@ -99,17 +97,20 @@ func (h *ContextHandler) Middleware(next http.Handler) http.Handler {
 			Logger:         log.New("context"),
 		}
 
-		// Inject ReqContext into http.Request.Context
-		*r = *r.WithContext(context.WithValue(ctx, reqContextKey{}, reqContext))
+		// inject ReqContext in the context
+		ctx = context.WithValue(ctx, reqContextKey{}, reqContext)
 		// store list of possible auth header in context
-		*reqContext.Req = *reqContext.Req.WithContext(WithAuthHTTPHeaders(reqContext.Req.Context(), h.Cfg))
+		ctx = WithAuthHTTPHeaders(ctx, h.Cfg)
+		// Set the context for the http.Request.Context
+		// This modifies both r and reqContext.Req since they point to the same value
+		*reqContext.Req = *reqContext.Req.WithContext(ctx)
 
-		traceID := tracing.TraceIDFromContext(mContext.Req.Context(), false)
+		traceID := tracing.TraceIDFromContext(reqContext.Req.Context(), false)
```

Comment:
@gamab why is this not the ctx.Req.Context().

I guess you refactored how we populate. just wanted to make sure this is correct

Reply:
It could have been `r.Context()` or `reqContext.Req.Context()` they are equivalent as they point to the same thing.
I think we could have used `ctx` straight away as well as it has the traceID.
But we decided to stay close to what was done previously.